lib/service_check.pm

```diff
@@ -140,15 +165,44 @@ sub install_services {
 
 sub check_services {
     my $service = shift;
+    my $service_func;
     foreach my $s (keys %$service) {
         my $srv_proc_name = $service->{$s}->{srv_proc_name};
         my $support_ver   = $service->{$s}->{support_ver};
         if (grep { $_ eq $hdd_base_version } split(',', $support_ver)) {
+            # service check after migration. if the we've set up service check
+            # function, we don't need following actions to check the service.
+            if (exists $service->{$s}->{service_check_func}) {
+                $service_func = $service->{$s}->{service_check_func};
+                call_func(\&$service_func, 'after');
+                next;
+            }
+
             assert_script_run 'systemctl status '
               . $srv_proc_name
-              . ' --no-pager | grep active';
+              . ' --no-pager | grep -w active';
         }
     }
 }
 
+#
+# check apache service before and after migration
+#
+sub apache_service_check {
```

Comment:
1. you have 2 functions having exactly same words in name but put in different order `apache_service_check` and `check_apache_service`. unless you are author of this functions no chance to figure out which one is doing what .
You may come up with different name but I would vote for `full_apache_check`
2. this function is just gather functions declared in http_srv I think more clear would be move it to http_srv and also replace duplicate code in run function of http_srv with ( using current name to not confuse you )  `apache_service_check('before')`

Reply:
Agree. We are duplicating code in both places. I would move the logic to http_srv.pm . 

In any case, if this grows with other services and the logic is similar, we can come up with a generic way of calling same functions for each service.

Reply:
thanks for the comments, these functions are we listed in the prototype we discussed before,
I'll check with our team for better solution.

Reply:
@rwx788 @Soulofdestiny What's your opinion on "move the logic to http_srv.pm" as jlausuch  suggest?

Reply:
For me it looks like in this PR we have things which are in test module but should be in the lib and vice versa, which is confusing.